raiden/smart_contracts/TokenNetwork.sol

```diff
@@ -0,0 +1,563 @@
+    mapping (uint => ClosingRequest) public closing_requests;
+
+    /*
+     *  Data Structures
+     */
+
+    struct Participant
+    {
+        // Total amount of token transferred to this smart contract through the
+        // `deposit` function, note that direct token transfer cannot be
+        // tracked and will be burned.
+        uint256 balance;
+
+        // The latest known merkle root of the pending hash-time locks, used to
+        // validate the withdrawn proofs.
+        bytes32 locksroot;
+
+        // The latest known transferred_amount from this node to the other
+        // participant, used to compute the net balance on settlement.
+        uint256 transferred_amount;
+
+        // Value used to order transfers and only accept the latest on calls to
+        // update, this will only be relevant after either #182 or #293 is
+        // implemented.
+        uint64 nonce;
+
+        // This value is set to True after the channel has been opened.
+        // This is an efficient way to mark the channel participants without doing a deposit.
+        // This is uint8 and it gets packed with the nonce.
+        bool initialized;
+
+        // A mapping to keep track of locks that have been withdrawn.
+        mapping(bytes32 => bool) withdrawn_locks;
+    }
+
+    struct Channel {
+        // We also use the settle_timeout to verify that
+        // the channel is open
+        uint settle_timeout;
+        mapping(address => Participant) participants;
+    }
+
+    struct ClosingRequest {
+        address closing_address;
+
+        // Block number at which the settlement window ends.
+        uint settle_block_number;
+    }
+
+    /*
+     *  Events
+     */
+
+    event ChannelOpened(
+        uint channel_identifier,
+        address participant1,
+        address participant2,
+        uint settle_timeout
+    );
+
+    event ChannelNewBalance(uint channel_identifier, address participant, uint balance);
+
+    event ChannelClosed(uint channel_identifier, address closing_address);
+
+    event ChannelUnlocked(uint channel_identifier, address participant, uint transferred_amount);
+
+    event TransferUpdated(uint channel_identifier, address participant);
+
+    event ChannelSettled(uint channel_identifier);
+
+    /*
+     * Modifiers
+     */
+
+    modifier isClosed(uint channel_identifier) {
+        require(closing_requests[channel_identifier].settle_block_number > 0);
+        _;
+    }
+
+    modifier stillTimeout(uint channel_identifier) {
+        require(closing_requests[channel_identifier].settle_block_number > block.number);
+        _;
+    }
+
+    modifier timeoutOver(uint channel_identifier) {
+        require(closing_requests[channel_identifier].settle_block_number <= block.number);
+        _;
+    }
+
+    // Note: we use the settleTimeout to check if the channel is open. It must always be > 0.
+    modifier settleTimeoutValid(uint timeout) {
+        require(timeout >= 6 && timeout <= 2700000);
```

Comment:
Comment where the numbers come from

Reply:
No, this is a discussion point (added in the description). I just reused what was before.